kie-maven-plugin/src/main/java/org/kie/maven/plugin/GeneratePMMLModelMojo.java

```diff
@@ -0,0 +1,322 @@
+                urls.add(new File(element).toURI().toURL());
+            }
+
+            project.setArtifactFilter(new CumulativeScopeArtifactFilter(Arrays.asList("compile",
+                                                                                      "runtime")));
+            for (Artifact artifact : project.getArtifacts()) {
+                File file = artifact.getFile();
+                if (file != null) {
+                    urls.add(file.toURI().toURL());
+                }
+            }
+            urls.add(outputDirectory.toURI().toURL());
+
+            ClassLoader projectClassLoader = URLClassLoader.newInstance(urls.toArray(new URL[0]),
+                                                                        getClass().getClassLoader());
+            Thread.currentThread().setContextClassLoader(projectClassLoader);
+        } catch (DependencyResolutionRequiredException | MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+
+        KieServices ks = KieServices.Factory.get();
+        try {
+            // FLAG TO KNOW THAT BUILD HAS BEEN LAUNCHED BY MAVEN
+            properties.put("kie-maven-plugin-launcher", "true");
+            setSystemProperties(properties);
+
+            final KieBuilderImpl kieBuilder = (KieBuilderImpl) ks.newKieBuilder(projectDir);
+            kieBuilder.setPomModel(new ProjectPomModel(mavenSession));
+            String testPath = "src" + File.separator + "test" + File.separator + "java";
+            String pmmlResourcePath = "src" + File.separator + "main" + File.separator + "resources" + File.separator + PMMLRESOURCES;
+            kieBuilder.buildAll(ExecutableModelMavenProject.SUPPLIER,
+                                s -> !s.contains(testPath) && !s.contains(pmmlResourcePath));
+            InternalKieModule kieModule = (InternalKieModule) kieBuilder.getKieModule();
+            List<String> generatedFiles = kieModule.getFileNames()
+                    .stream()
+                    .filter(f -> f.endsWith("java"))
+                    .collect(Collectors.toList());
+
+            Set<String> drlFiles = kieModule.getFileNames()
+                    .stream()
+                    .filter(f -> f.endsWith("drl"))
+                    .collect(Collectors.toSet());
+
+            getLog().info(String.format("Found %d generated files in Canonical Model", generatedFiles.size()));
+
+            MemoryFileSystem mfs = kieModule instanceof CanonicalKieModule ?
+                    ((MemoryKieModule) ((CanonicalKieModule) kieModule).getInternalKieModule()).getMemoryFileSystem() :
+                    ((MemoryKieModule) kieModule).getMemoryFileSystem();
+
+            final String droolsModelCompilerPath = "/generated-sources/drools-model-compiler/main/java";
+            final String newCompileSourceRoot = targetDirectory.getPath() + droolsModelCompilerPath;
+            project.addCompileSourceRoot(newCompileSourceRoot);
```

Comment:
What if this path doesn't exist? Is it created on the fly or it fail?

Reply:
@danielezonca 
that path is created later